### Kaabachi Alaa
### 3BD1

#### Étape 1 : Charger et examiner les données

In [10]:
# Charger les bibliothèques nécessaires
library(readxl)
library(nnet)
library(caret)

#Charger Les données de Exercice1
data <- read_excel("C:\\Users\\alaak\\Downloads\\exercice 1.xlsx", sheet="Feuil1")
#Afficher les premieéres lignes pour examiner les données
head(data)
#Vérifier les valuers manquantes
colSums(is.na(data))
#Supprimer les lignes avec des valeurs manquantes
data <- na.omit(data)


ID sexe Region Age etat_civil N_educatif Membre_famille Revenu_famille
1 1  0    1      2   1          3           5             0             
2 2  0    1      4   2          3           5             0             
3 3  0    1      4   2          3          11             0             
4 5  0    1      3   2          3           4             0             
5 8  0    1      1   1          3           5             1             
6 9  0    1      4   3          3           4             1             
  Revenu_Mensuelle Redoublant ⋯ Niveau_educ_pere Niveau_educ_mere Sign_ecole
1   0              1          ⋯ 2                2                1         
2   0              1          ⋯ 2                1                1         
3 100              1          ⋯ 2                1                2         
4   0              1          ⋯ 1                1                1         
5   0              1          ⋯ 3                3                3         
6 100              0          ⋯ 2                3                4         
  Endroit_freq Ph_famille Ricule_Prof Ricule_collegue Sensation_apres_ecole
1 1            1          0           1               1                    
2 2            2          1           1               1                    
3 1            3          0           0               1                    
4 1            2          1           1               3                    
5 4            4          0           0               3                    
6 1            4          1           1               1                    
  Apres_repture_ecole Violence
1 1                   1       
2 1                   1       
3 2                   0       
4 3                   1       
5 2                   1       
6 2                   0

ID                  sexe                Region 
                    0                     0                     0 
                  Age            etat_civil            N_educatif 
                    0                     0                     0 
       Membre_famille        Revenu_famille      Revenu_Mensuelle 
                    0                     0                     0 
           Redoublant        Prob_Enseignat            Aide_ecole 
                    0                     0                     0 
       Rq_Comprtement          Relat_Parent      Niveau_educ_pere 
                    0                     1                     0 
     Niveau_educ_mere            Sign_ecole          Endroit_freq 
                    0                     0                     0 
           Ph_famille           Ricule_Prof       Ricule_collegue 
                    0                     0                     0 
Sensation_apres_ecole   Apres_repture_ecole              Violence 
                    0                     0                     0

### 2. Expliquer les variables a variable après rupture école en fonction des variables


In [11]:
model <- multinom(Apres_repture_ecole ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model)

# weights:  88 (63 variable)
initial  value 124.766493 
iter  10 value 78.622797
iter  20 value 63.239268
iter  30 value 55.258599
iter  40 value 49.846960
iter  50 value 47.101855
iter  60 value 43.719936
iter  70 value 43.481027
iter  80 value 43.465618
iter  90 value 43.465149
final  value 43.465144 
converged


Warning message in sqrt(diag(vc)):
"NaNs produced"


Call:
multinom(formula = Apres_repture_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
  (Intercept)      sexe         Age       Region  etat_civil    N_educatif
2   -7.489772 245.87263  -0.4376716   -2.4084838    4.901879 -5.771419e-01
3  -10.469941  47.08085   0.7066992    0.6411367    5.941687  8.525498e-03
4 -611.468711  82.99164 208.3453744 -225.7999524 -128.520080 -1.605677e+02
  Membre_famille Revenu_famille Revenu_Mensuelle   Redoublant Prob_Enseignat
2      0.2134464      0.4469855       0.00356250    0.6462154      1.3745384
3      0.1245541      0.7613484       0.00124591   -0.2674273     -0.3408284
4     23.4909023    -32.9186956       0.42818081 -187.7576202    -38.2412999
  Aide

### interprétation
#### sexe :
Pour la classe 2 : Coefficient très élevé (245.87), ce qui signifie que cette variable a un effet important pour prédire l'appartenance à cette classe. Cela peut indiquer une forte dépendance entre le genre et la classe 2.
Pour la classe 3 : Coefficient élevé mais plus modéré (47.08).
Pour la classe 4 : Valeur extrême (82.99), suggérant une influence importante, mais cela pourrait aussi refléter des déséquilibres ou des anomalies dans les données.
#### Age :
Pour la classe 2 : Coefficient négatif (-0.43), ce qui signifie que plus l’âge augmente, moins il est probable d’appartenir à cette classe.
Pour la classe 3 : Coefficient positif (0.70), donc une augmentation de l’âge favorise cette classe.
Pour la classe 4 : Coefficient extrêmement élevé (208.34), mais cela pourrait être dû à une surreprésentation ou un biais.
#### Region :
Pour la classe 2 : Coefficient négatif (-2.40), indiquant que cette région réduit la probabilité d’appartenir à cette classe.
Pour la classe 3 : Coefficient modéré et positif (0.64).
Pour la classe 4 : Coefficient fortement négatif (-225.79), suggérant une très faible probabilité.
#### etat_civil (État civil) :
Pour la classe 2 et 3 : Effets modérés et positifs (4.90 et 5.94 respectivement), donc certains états civils augmentent la probabilité d’appartenir à ces classes.
Pour la classe 4 : Coefficient fortement négatif (-128.52), ce qui diminue fortement cette probabilité.
#### N_educatif (Niveau éducatif) :
Les effets sont relativement faibles pour les classes 2 et 3 (-0.57 et 0.0085, proches de 0).
Pour la classe 4 : Coefficient fortement négatif (-160.56), indiquant une influence importante.
#### Membre_famille :
Pour toutes les classes, l’effet est faible, avec des coefficients proches de 0 (entre 0.12 et 23.49 pour la classe 4).
#### Revenu_famille :
Classes 2 et 3 : Coefficients faibles (0.44 et 0.76), donc faible impact.
Classe 4 : Coefficient négatif élevé (-32.91), indiquant un effet important sur la réduction de probabilité.
#### Redoublant :
Classe 2 : Faible effet positif (0.64).
Classe 3 : Faible effet négatif (-0.26).
Classe 4 : Effet négatif très important (-187.75), ce qui signifie que le redoublement est fortement lié à une moindre probabilité pour cette classe.
#### Rq_Comprtement (Remarques sur le comportement) :
Classe 2 et 3 : Effets modérés négatifs (-1.24 et -0.87), donc un mauvais comportement réduit la probabilité d’appartenance.
Classe 4 : Effet positif élevé (110.97), ce qui pourrait suggérer que certaines remarques sont liées à cette classe.
#### Sign_ecole (Signes liés à l'école) :
Classe 2 : Faible effet positif (0.25).
Classe 3 : Effet modéré positif (1.06).
Classe 4 : Coefficient élevé (50.50), donc impact important pour cette classe.
#### Ricule_Prof (Moqueries du professeur) :
Classe 2 : Coefficient négatif (-1.19), donc réduit la probabilité.
Classe 3 : Coefficient très négatif (-3.50), indiquant un effet encore plus fort.
Classe 4 : Coefficient extrêmement positif (178.04), suggérant une association importante.
#### Ricule_collegue (Moqueries des collègues) :
Classe 2 : Faible effet positif (0.81).
Classe 3 : Coefficient positif modéré (2.18).
Classe 4 : Coefficient extrêmement négatif (-268.08), ce qui réduit fortement la probabilité.

### 3.Determiner la Qualité du modéle

In [19]:
#D'abord on va faire une prédiction
predictions<-predict(model,newdata=data)
#creer la matrice de confusion
MatriceDeConfusion<-table(data$Apres_repture_ecole,predictions)
print(MatriceDeConfusion)
#Calculer la qualité du modéle:somme de diagonale / somme de matrice
print(paste("Qualité du modèle :", sum(diag(MatriceDeConfusion)) / sum(MatriceDeConfusion)))

   predictions
     1  2  3  4
  1 29  0  0  0
  2  3 23  5  0
  3  2  5 12  1
  4  0  0  0 10
[1] "Qualité du modèle : 0.822222222222222"


### 4. Expliquer la variable sensation après rupture école en fonction des variablesµ

In [22]:
model <- multinom(Sensation_apres_ecole ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model)

# weights:  66 (42 variable)
initial  value 98.875106 
iter  10 value 69.921757
iter  20 value 63.545892
iter  30 value 62.811185
iter  40 value 62.710519
iter  50 value 62.693622
iter  60 value 62.691463
final  value 62.691445 
converged


Call:
multinom(formula = Sensation_apres_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
  (Intercept)       sexe        Age    Region etat_civil N_educatif
2  -2.0092479 -15.288989 1.70302770  2.247275 -2.1451832 -1.1150235
3  -0.9491642   1.817137 0.05406014 -1.968661  0.4986185 -0.6205035
  Membre_famille Revenu_famille Revenu_Mensuelle Redoublant Prob_Enseignat
2    -0.06729289     -1.5735217     0.0023857860 -0.5856426     -0.1390584
3    -0.14471002     -0.3600626     0.0007416636  3.8445603      1.6831663
  Aide_ecole Rq_Comprtement Relat_Parent Niveau_educ_pere Niveau_educ_mere
2 -0.1679547     0.14149738  -0.33008611        -0.311379        0.9435157
3  0.3077777    -0.02068932   0.0


#### sexe
Variable qui a un effet très fort sur les résultats de la prédiction.
Classe 2 : Coefficient 245.87 → Cela suggère que cette variable favorise considérablement l'appartenance à la classe 2.
Classe 3 : Coefficient 47.08 → Influence positive, mais moins forte que pour la classe 2.
Classe 4 : Coefficient 82.99 → Influence également positive, mais encore plus faible que pour la classe 2.
#### Age
L'âge a une influence modérée sur les résultats.
Classe 2 : Coefficient -0.44 → Plus l’âge augmente, moins il est probable d’appartenir à la classe 2.
Classe 3 : Coefficient 0.71 → Augmentation de l'âge favorise l’appartenance à la classe 3.
Classe 4 : Coefficient 208.35 → L'augmentation de l'âge favorise fortement l'appartenance à la classe 4.
#### Region
L'effet de la région peut être significatif selon la classe.
Classe 2 : Coefficient -2.41 → Les régions spécifiques diminuent la probabilité d’appartenir à la classe 2.
Classe 3 : Coefficient 0.64 → L'effet est positif dans cette classe.
Classe 4 : Coefficient -225.80 → Effet fortement négatif, indiquant une probabilité beaucoup plus faible dans cette région.
#### etat_civil (État civil)
L'état civil affecte les probabilités d'appartenance aux différentes classes.
Classe 2 : Coefficient 4.90 → L'état civil influence positivement la probabilité d’appartenir à la classe 2.
Classe 3 : Coefficient 5.94 → Influence similaire mais plus forte pour la classe 3.
Classe 4 : Coefficient -128.52 → Influence négative, réduisant fortement la probabilité d’appartenir à la classe 4.
#### N_educatif (Niveau éducatif)
L'influence du niveau éducatif varie de manière significative selon la classe.
Classe 2 : Coefficient -0.58 → Effet négatif modéré.
Classe 3 : Coefficient 0.01 → Effet extrêmement faible.
Classe 4 : Coefficient -160.56 → Effet négatif fort, indiquant une réduction importante de probabilité.
#### Membre_famille
Effet modéré à faible selon la classe.
Classe 2 : Coefficient 0.21 → Effet positif modéré.
Classe 3 : Coefficient 0.12 → Effet très faible.
Classe 4 : Coefficient 23.49 → Influence significative et fortement positive pour la classe 4.
#### Revenu_famille
Effet variable selon la classe.
Classe 2 : Coefficient 0.44 → Augmentation du revenu familial favorise légèrement l’appartenance à cette classe.
Classe 3 : Coefficient 0.76 → L'effet est plus prononcé dans cette classe.
Classe 4 : Coefficient -32.92 → Effet négatif important, la probabilité d'appartenir à la classe 4 diminue fortement avec un revenu familial plus élevé.
#### Revenu_Mensuelle (Revenu Mensuel)
L'effet du revenu mensuel sur l'appartenance aux classes.
Classe 2 : Coefficient 0.0036 → Effet positif faible.
Classe 3 : Coefficient 0.0012 → Effet positif mais moins significatif.
Classe 4 : Coefficient 0.43 → Effet modéré à fort dans cette classe.
#### Redoublant (Redoublement)
Le fait d’être un redoublant joue un rôle important.
Classe 2 : Coefficient 0.65 → Influence positive sur la probabilité d’appartenir à la classe 2.
Classe 3 : Coefficient -0.27 → Effet négatif, réduisant la probabilité d'appartenir à la classe 3.
Classe 4 : Coefficient -187.76 → Effet extrêmement négatif, réduisant fortement la probabilité d'appartenir à la classe 4.
#### Prob_Enseignat (Probabilité d'enseignement)
La probabilité d'enseignement affecte les résultats.
Classe 2 : Coefficient 1.37 → L’augmentation de la probabilité d’enseignement favorise l’appartenance à cette classe.
Classe 3 : Coefficient -0.34 → L'effet est négatif dans cette classe.
Classe 4 : Coefficient -38.24 → Effet négatif extrêmement fort.
#### Aide_ecole (Aide scolaire)
Influence modérée sur les résultats.
Classe 2 : Coefficient 0.52 → Augmentation de l'aide scolaire favorise l'appartenance à cette classe.
Classe 3 : Coefficient 0.55 → Effet légèrement plus prononcé.
Classe 4 : Coefficient 62.25 → Effet extrêmement positif pour la classe 4.
#### Rq_Comprtement (Remarques comportementales)
Le comportement a un impact négatif modéré.
Classe 2 : Coefficient -1.24 → Influence négative sur la probabilité d’appartenir à la classe 2.
Classe 3 : Coefficient -0.87 → Effet également négatif mais plus faible.
Classe 4 : Coefficient 110.98 → Effet positif extrêmement fort.
#### Relat_Parent (Relation avec les parents)
La relation avec les parents semble avoir un impact significatif.
Classe 2 : Coefficient -0.67 → Influence négative.
Classe 3 : Coefficient -0.72 → Effet négatif plus prononcé.
Classe 4 : Coefficient -90.31 → Effet très négatif, indiquant une probabilité fortement réduite.
#### Niveau_educ_pere (Niveau éducatif du père)
L’influence du niveau éducatif du père sur la prédiction.
Classe 2 : Coefficient 0.41 → Effet positif modéré.
Classe 3 : Coefficient -1.10 → Effet négatif.
Classe 4 : Coefficient 175.69 → Forte influence positive pour la classe 4.
#### Niveau_educ_mere (Niveau éducatif de la mère)
Effet similaire à celui du père mais plus significatif.
Classe 2 : Coefficient 0.99 → Effet positif modéré.
Classe 3 : Coefficient -0.10 → Effet légèrement négatif.
Classe 4 : Coefficient 14.70 → Influence très positive.
#### Sign_ecole (Signification de l'école)
L’importance de l'école dans le processus éducatif.
Classe 2 : Coefficient 0.25 → Effet positif.
Classe 3 : Coefficient 1.06 → Effet plus important.
Classe 4 : Coefficient 50.51 → Influence extrêmement positive.
#### Endroit_freq (Endroit fréquenté)
L'endroit fréquenté par l'individu affecte les probabilités.
Classe 2 : Coefficient 0.14 → Influence positive modérée.
Classe 3 : Coefficient -0.43 → Effet négatif dans cette classe.
Classe 4 : Coefficient -9.84 → Effet négatif très important.
#### Ph_famille (Phénomène familial)
L'impact des phénomènes familiaux sur l’appartenance à une classe.
Classe 2 : Coefficient 0.15 → Effet modéré.
Classe 3 : Coefficient -0.04 → Effet faible.
Classe 4 : Coefficient -4.39 → Effet négatif significatif.


### 5.Determiner la Qualité du modéle

In [23]:
#D'abord on va faire une prédiction
predictions<-predict(model,newdata=data)
#creer la matrice de confusion
MatriceDeConfusion<-table(data$Sensation_apres_ecole,predictions)
print(MatriceDeConfusion)
#Calculer la qualité du modéle:somme de diagonale / somme de matrice
print(paste("Qualité du modèle :", sum(diag(MatriceDeConfusion)) / sum(MatriceDeConfusion)))

   predictions
     1  2  3
  1 21  4  7
  2  3 23  6
  3  3  6 17
[1] "Qualité du modèle : 0.677777777777778"


### 6. Expliquer la variable Violence en fonction des variables


In [24]:
model <- multinom(Violence ~ sexe + Age + Region + etat_civil
                   + N_educatif + Membre_famille + Revenu_famille
                   + Revenu_Mensuelle + Redoublant + Prob_Enseignat
                   + Aide_ecole + Rq_Comprtement + Relat_Parent
                   + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole
                   + Endroit_freq + Ph_famille + Ricule_Prof
                   + Ricule_collegue, data = data)
summary(model)

# weights:  22 (21 variable)
initial  value 62.383246 
iter  10 value 36.964839
iter  20 value 34.414180
final  value 34.343438 
converged


Call:
multinom(formula = Violence ~ sexe + Age + Region + etat_civil + 
    N_educatif + Membre_famille + Revenu_famille + Revenu_Mensuelle + 
    Redoublant + Prob_Enseignat + Aide_ecole + Rq_Comprtement + 
    Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole + 
    Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = data)

Coefficients:
                       Values   Std. Err.
(Intercept)       6.068413312 3.713088928
sexe              1.524809931 2.123160842
Age              -1.055798304 0.533031905
Region            1.002753591 1.025679325
etat_civil        0.023586691 0.854480858
N_educatif       -0.673341017 0.624908213
Membre_famille   -0.061502670 0.193547317
Revenu_famille    1.282931321 0.974624342
Revenu_Mensuelle -0.004960213 0.002147526
Redoublant       -1.211504998 0.971903269
Prob_Enseignat   -2.011345947 0.982459715
Aide_ecole       -0.196828062 0.270754003
Rq_Comprtement    0.509204931 0.424193473
Relat_Parent      0.418167382 0.67160

#### sexe
- Valeur : 1.5248
- Erreur standard : 2.1232
- L'effet du sexe est positif, ce qui suggère que les individus d'un sexe particulier (probablement masculin) ont une probabilité plus élevée d'appartenir à la classe cible. Cependant, l'erreur standard élevée indique que l'incertitude autour de cette estimation est assez grande, ce qui signifie que cet effet n'est pas très fiable.

#### Age
- Valeur : -1.0558
- Erreur standard : 0.5330
- L'âge a un effet négatif sur la probabilité d'appartenir à la classe cible. Cela signifie que plus une personne est âgée, moins elle est susceptible de faire partie de cette classe. L'erreur standard est relativement faible, ce qui signifie que cet effet est plus précis que celui de la variable sexe.

#### Region
- Valeur : 1.0028
- Erreur standard : 1.0257
- L'effet de la région est positif, ce qui suggère que les individus vivant dans certaines régions ont une probabilité plus élevée d'appartenir à la classe cible. Toutefois, l'incertitude autour de cette estimation est relativement élevée.

#### etat_civil
- Valeur : 0.0236
- Erreur standard : 0.8545
- L'effet de l'état civil est faible et positif, indiquant que l'état civil a peu d'impact sur la probabilité d'appartenir à la classe cible. L'erreur standard élevée signifie que cet effet est peu fiable.

#### N_educatif
- Valeur : -0.6733
- Erreur standard : 0.6249
- L'éducation a un effet négatif sur la probabilité d'appartenir à la classe cible. Les individus ayant un niveau éducatif plus élevé ont moins de chances d'appartenir à la classe cible, bien que l'erreur standard soit relativement élevée, ce qui suggère une certaine incertitude.

#### Membre_famille
- Valeur : -0.0615
- Erreur standard : 0.1935
- L'effet du nombre de membres dans la famille est faible et négatif, ce qui signifie que plus une personne a de membres dans sa famille, moins elle est susceptible d'appartenir à la classe cible. L'erreur standard est relativement faible, ce qui suggère que cet effet est plus fiable.

#### Revenu_famille
- Valeur : 1.2829
- Erreur standard : 0.9746
- Un revenu familial plus élevé est associé à une probabilité accrue d'appartenir à la classe cible. Cet effet est significatif, mais l'erreur standard relativement élevée indique qu'il existe une certaine incertitude.

#### Revenu_Mensuelle
- Valeur : -0.00496
- Erreur standard : 0.0021
- L'effet du revenu mensuel est négatif, mais très faible. Cela signifie qu'une augmentation du revenu mensuel réduit légèrement la probabilité d'appartenir à la classe cible. L'erreur standard est faible, ce qui suggère que cet effet est relativement précis.

#### Redoublant
- Valeur : -1.2115
- Erreur standard : 0.9719
- L'effet du fait d'avoir redoublé est négatif, indiquant que les individus ayant redoublé sont moins susceptibles d'appartenir à la classe cible. L'erreur standard relativement élevée montre que l'incertitude autour de cet effet est assez grande.

#### Prob_Enseignat
- Valeur : -2.0113
- Erreur standard : 0.9825
- Le fait d'être enseignant a un effet négatif significatif sur la probabilité d'appartenir à la classe cible. Les enseignants ont moins de chances d'appartenir à la classe cible, avec un effet relativement précis, vu l'erreur standard faible.

#### Aide_ecole
- Valeur : -0.1968
- Erreur standard : 0.2708
- L'effet de l'aide scolaire est légèrement négatif, suggérant que l'aide scolaire réduit légèrement la probabilité d'appartenir à la classe cible. Cependant, l'effet est faible et l'incertitude est modérée.

#### Rq_Comprtement
- Valeur : 0.5092
- Erreur standard : 0.4242
- Une gestion du comportement positive est associée à une probabilité accrue d'appartenir à la classe cible. Bien que cet effet soit positif, l'erreur standard indique que l'incertitude est assez importante.

#### Relat_Parent
- Valeur : 0.4182
- Erreur standard : 0.6716
- De bonnes relations avec les parents sont associées à une probabilité plus élevée d'appartenir à la classe cible, bien que l'incertitude autour de cet effet soit relativement importante.

#### Niveau_educ_pere
- Valeur : -0.5041
- Erreur standard : 0.7345
- Un niveau d'éducation plus élevé du père est associé à une probabilité réduite d'appartenir à la classe cible. Cet effet est négatif, mais l'incertitude est importante, comme l'indique l'erreur standard.

#### Niveau_educ_mere
- Valeur : -0.2043
- Erreur standard : 0.6725
- Le niveau d'éducation de la mère a également un effet négatif sur la probabilité d'appartenir à la classe cible, bien que cet effet soit faible et l'incertitude relativement élevée.

#### Sign_ecole
- Valeur : -0.0940
- Erreur standard : 0.3747
- Les signaux provenant de l'école ont un effet légèrement négatif, mais cet effet est très faible et l'incertitude est relativement élevée.

#### Endroit_freq
- Valeur : 0.3923
- Erreur standard : 0.2744
- L'effet positif de la fréquence du lieu indique que des fréquentes visites ou interactions avec l'endroit en question augmentent la probabilité d'appartenir à la classe cible. L'effet est modéré et l'incertitude est relativement faible.

#### Ph_famille
- Valeur : -1.1593
- Erreur standard : 0.4205
- Les problèmes familiaux ont un effet négatif important, ce qui signifie que des problèmes familiaux importants réduisent la probabilité d'appartenir à la classe cible. L'effet est significatif avec une faible erreur standard.

#### Ricule_Prof
- Valeur : 2.8296
- Erreur standard : 1.2293
- L'effet positif du ridicule par les professeurs suggère que des comportements négatifs de la part des professeurs augmentent les chances d'appartenir à la classe cible. L'erreur standard montre qu'il existe une incertitude modérée autour de cette estimation.

#### Ricule_collegue
- Valeur : 0.9835
- Erreur standard : 0.8978
- L'effet positif du ridicule par les collègues montre que des comportements négatifs de la part des collègues augmentent les chances d'appartenir à la classe cible. Cependant, l'incertitude est relativement élevée en raison de l'erreur standard.



#### 7.Determiner la Qualité du modéle

In [25]:
#D'abord on va faire une prédiction
predictions<-predict(model,newdata=data)
#creer la matrice de confusion
MatriceDeConfusion<-table(data$Violence,predictions)
print(MatriceDeConfusion)
#Calculer la qualité du modéle:somme de diagonale / somme de matrice
print(paste("Qualité du modèle :", sum(diag(MatriceDeConfusion)) / sum(MatriceDeConfusion)))

   predictions
     0  1
  0 54  6
  1 12 18
[1] "Qualité du modèle : 0.8"
